compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -907,18 +911,32 @@ private BLangStatementExpression createIfElseFromConfigurable(BLangSimpleVariabl
                     // Module init should fail if listener is a error value.
                     if (Symbols.isFlagOn(globalVarFlags, Flags.LISTENER)
                             && types.containsErrorType(globalVar.expr.type)) {
-                        globalVar.expr = ASTBuilderUtil.createCheckExpr(globalVar.expr.pos, globalVar.expr, globalVar.type);
+                        globalVar.expr = ASTBuilderUtil.createCheckExpr(globalVar.expr.pos, globalVar.expr,
+                                globalVar.type);
                     }
 
                     addToInitFunction(simpleGlobalVar, initFnBody);
                     desugaredGlobalVarList.add(simpleGlobalVar);
                     break;
             }
-        });
+        }
+
         this.env.enclPkg.topLevelNodes.addAll(desugaredGlobalVarList);
         return desugaredGlobalVarList;
     }
 
+    private void addToGlobalVariableList(BLangStatement bLangStatement, BLangBlockFunctionBody initFnBody,
+                                         BLangVariable globalVar, List<BLangVariable> desugaredGlobalVarList) {
+        BLangSimpleVariable simpleVar = ((BLangSimpleVariableDef) bLangStatement).var;
+        if ((simpleVar.symbol.owner.tag & SymTag.PACKAGE) != SymTag.PACKAGE) {
+            initFnBody.stmts.add(bLangStatement);
+        } else {
```

Comment:
If we return in the if block we can remove the else block and move the logic out.

Reply:
I think else block is much cleaner here right?

Reply:
Doesn't make much of a difference here, right? As a practice, we usually try and return early and remove the else blocks.

Reply:
But yeah this is subjective. :smiley:

Reply:
Removed else block